src/main/scala/net.manub.embeddedkafka/schemaregistry/package.scala

```diff
@@ -1,8 +1,8 @@
 package net.manub.embeddedkafka
 
-import io.confluent.kafka.serializers.{
-  KafkaAvroDeserializer => ConfluentKafkaAvroDeserializer,
-  KafkaAvroSerializer => ConfluentKafkaAvroSerializer
+import io.confluent.kafka.streams.serdes.avro.{
+  GenericAvroSerde,
```

Comment:
@kmatasflp Yes, they do use `io.confluent.kafka.serializers.KafkaAvroSerializer` under the hood, however, if we want to use KafkaAvroSerializer directly, we'd have to pretty much re-implement the same thing here https://github.com/confluentinc/schema-registry/tree/master/avro-serde/src/main/java/io/confluent/kafka/streams/serdes/avro 
I'd rather reuse the existing code provided by an official confluent library instead of rewriting my own. 

That being said, the above library does not depend on kafka-streams according here https://github.com/confluentinc/schema-registry/blob/master/avro-serde/pom.xml#L41
It might not be the best idea to add `kafka.streams` in the namespace, but let's leave it to confluent to figure out.

Reply:
I do not understand what do you need to reimplement here?

Reply:
Following works: 

```
package net.manub.embeddedkafka

import io.confluent.kafka.serializers.{KafkaAvroDeserializer => ConfluentKafkaAvroDeserializer, KafkaAvroSerializer => ConfluentKafkaAvroSerializer}
import net.manub.embeddedkafka.schemaregistry.EmbeddedKafka.{configForSchemaRegistry, consumerConfigForSchemaRegistry}
import net.manub.embeddedkafka.schemaregistry.{EmbeddedKafkaConfig => EmbeddedKafkaSRConfig}
import org.apache.avro.generic.GenericRecord
import org.apache.avro.specific.SpecificRecord
import org.apache.kafka.common.serialization.{Deserializer, Serde, Serdes, Serializer}

import scala.collection.JavaConverters._

package object schemaregistry {

  private def serdeFrom[T](serConfig: Map[String, Object], deserConfig: Map[String, Object], isKey: Boolean)= {
    val ser = new ConfluentKafkaAvroSerializer
    ser.configure(serConfig.asJava, isKey)
    val deser = new ConfluentKafkaAvroDeserializer
    deser.configure(deserConfig.asJava, isKey)

    Serdes.serdeFrom(ser, deser).asInstanceOf[Serde[T]]
  }

  //SpecificRecord

  def specificSerdeFrom[T <: SpecificRecord](isKey: Boolean)(
      implicit config: EmbeddedKafkaSRConfig): Serde[T] = {
    serdeFrom[T](configForSchemaRegistry, consumerConfigForSchemaRegistry, isKey)
  }

  implicit def valueSpecificAvroSerdeFrom[T <: SpecificRecord](
      implicit config: EmbeddedKafkaSRConfig): Serde[T] = {
    specificSerdeFrom[T](isKey = false)
  }

  implicit def valueSpecificAvroSerializer[T <: SpecificRecord](
      implicit config: EmbeddedKafkaSRConfig): Serializer[T] =
    specificSerdeFrom[T](isKey = false).serializer

  implicit def valueSpecificAvroDeserializer[T <: SpecificRecord](
      implicit config: EmbeddedKafkaSRConfig): Deserializer[T] =
    specificSerdeFrom[T](isKey = false).deserializer

  // GenericRecord

  def genericSerde(isKey: Boolean)(
      implicit config: EmbeddedKafkaSRConfig) = {
    serdeFrom[GenericRecord](configForSchemaRegistry, configForSchemaRegistry, isKey)
  }

  implicit def valueGenericAvroSerde(
      implicit config: EmbeddedKafkaSRConfig): Serde[GenericRecord] = {
    genericSerde(isKey = false)
  }

  implicit def valueGenericAvroSerializer(
      implicit config: EmbeddedKafkaSRConfig): Serializer[GenericRecord] =
    genericSerde(isKey = false).serializer

  implicit def valueGenericAvroDeserializer(
      implicit config: EmbeddedKafkaSRConfig): Deserializer[GenericRecord] =
    genericSerde(isKey = false).deserializer
}

```

On has to use `consumerConfigForSchemaRegistry` when deserializing specific record and `configForSchemaRegistry` when deserializing generic record. Also made explicit that the `Serdes` one gets from here are for values not for keys, exposed factory method for creating `Serdes` for keys/values

Reply:
Good catch on the `configForSchemaRegistry` and `consumerConfigForSchemaRegistry` part. 👍

Reply:
Should it be changed to
```scala
def genericSerde(isKey: Boolean)(
      implicit config: EmbeddedKafkaSRConfig) = {
    serdeFrom[GenericRecord](configForSchemaRegistry, consumerConfigForSchemaRegistry, isKey)
  }
```?